src/Namshi/JOSE/JWS.php

```diff
@@ -24,7 +24,17 @@ class JWS extends JWT
      */
     public function __construct($algorithm, $type = null)
     {
-        parent::__construct(array(), array('alg' => $algorithm, 'typ' => $type ?: "JWS"));
+        if (is_string($algorithm)) {
+            $header = array('alg' => $algorithm, 'typ' => $type ?: "JWS");
+        }
+        else {
+            $header = $algorithm;
+            if (! isset($header['typ'])) {
+                $header['typ'] = $type ?: "JWS";
+            }
+        }
+
+        parent::__construct(array(), $header);
```

Comment:
Yeah, I'm okay with that, I was just trying to maintain backwards compatibility to prevent the big version bump.